src/internet_identity/internet_identity.did

```diff
@@ -333,42 +272,112 @@ type DeployArchiveResult = variant {
 };
 
 type BufferedArchiveEntry = record {
-    anchor_number: UserNumber;
+    anchor_number: IdentityNumber;
     timestamp: Timestamp;
     sequence_number: nat64;
     entry: blob;
 };
 
+// This is a placeholder result that will be replaced by the real result type
+// once we introduce actual errors. This is possible, because the result is
+// always used in an opt position.
+type result = variant {
+    ok;
+};
+
+type WebAuthnDirectSigMode = variant {
+    mandatory;
+    optional;
+};
+
+type key = variant {
```

Comment:
I would suggest using some other name than `key`, as "key" is a very generic and often overloaded term. For example in this specification it collides with `PublicKey` and `SessionKey`, which are not instances of `key` as defined in this line. 

Maybe something more specific to the actual functionality/abstraction would be more suitable, like `AuthnMethod` or `AuthnForm`?  I realise, that these are significantly longer than just "key" but they make it (more) clear what is represented by the structures.  As we move away from "anchor" to "identity", speaking of "identity's authentication method" is more clear than about "identity's key", no?

Another alternative could be a `pass`: an identity can have many "passes" (which allow the identity to authenticate itself).

Reply:
I find `auth` better than `key`, but my current preference would be `pass`:  the short for authentication is usually `authn`, like in `WebAuthn` (also to distinguish it from authorization, `authz`), so `auth` is a bit ambiguous, and even if we assume it refers to authentication, the connection to "method" is not obvious.  We could introduce e.g. `authm` for "authentication method" to disambiguate, but that's not perfect either...  
I like `pass` because it is short and describes "something that can be used to authenticate oneself", which I believe is what we are dealing with here, so we get a nice abstraction.
Maybe let's talk briefly together with @nmattia and settle on some naming (say this afternoon?), to avoid repeated renamings?

Reply:
I discussed with @nmattia yesterday:
* he suggested that we could also go for `passkey`, but since `passkeys` are strongly associated with WebAuthn (at least for now) I would like to go with something that conveys a more general meaning.
* he suggested also to change the method names by using prefixes in lieu of REST URL path segments: i.e. do `pass_update_settings`, `pass_add`, ... instead of of `add_pass`, `update_pass_settings`
* I'd be fine with using `pass` although I've have not seen it used so far as a synonym for "generic authentication method". I have slight preference for `authn` but I'm fine with `pass` too.